src/sagemaker/model.py

```diff
@@ -217,6 +218,9 @@ def deploy(self, initial_instance_count, instance_type, accelerator_type=None, e
                 For more information: https://docs.aws.amazon.com/sagemaker/latest/dg/ei.html
             endpoint_name (str): The name of the endpoint to create (default: None).
                 If not specified, a unique endpoint name will be created.
+            update_endpoint (bool): Flag to update the model in an existing Amazon SageMaker endpoint.
+                If True, this will deploy a new EndpointConfig to an already existing endpoint and delete resources
+                corresponding to the previous EndpointConfig. Default: False
```

Comment:
Please docstring what happens when this flag is false as well. Is it possible to throw an exception for example?

Reply:
Updated documentation. The error is raised in session.update_endpoint so the corresponding information is documented there.